pkg/operator/certrotation/client_cert_rotation_controller_test.go

```diff
@@ -0,0 +1,418 @@
+		Informer:              informerFactory.Core().V1().Secrets(),
+		EventRecorder:         eventRecorder,
+		AdditionalAnnotations: additionalAnnotations,
+		Owner:                 owner,
+		UseSecretUpdateOnly:   true,
+	}
+	caBundleConfigMap := CABundleConfigMap{
+		Namespace:             ns,
+		Name:                  caName,
+		Client:                client.CoreV1(),
+		Lister:                corev1listers.NewConfigMapLister(indexer),
+		EventRecorder:         eventRecorder,
+		Informer:              informerFactory.Core().V1().ConfigMaps(),
+		AdditionalAnnotations: additionalAnnotations,
+		Owner:                 owner,
+	}
+	targetFirstSecret := RotatedSelfSignedCertKeySecret{
+		Name:                  targetFirstName,
+		Namespace:             ns,
+		Validity:              24 * time.Hour,
+		Refresh:               12 * time.Hour,
+		CertCreator:           firstTargetCertCreator,
+		Client:                client.CoreV1(),
+		Informer:              informerFactory.Core().V1().Secrets(),
+		Lister:                corev1listers.NewSecretLister(indexer),
+		EventRecorder:         eventRecorder,
+		AdditionalAnnotations: additionalAnnotations,
+		Owner:                 owner,
+		UseSecretUpdateOnly:   true,
+	}
+	targetSecondSecret := RotatedSelfSignedCertKeySecret{
+		Name:                  targetSecondName,
+		Namespace:             ns,
+		Validity:              24 * time.Hour,
+		Refresh:               12 * time.Hour,
+		CertCreator:           secondTargetCertCreator,
+		Client:                client.CoreV1(),
+		Informer:              informerFactory.Core().V1().Secrets(),
+		Lister:                corev1listers.NewSecretLister(indexer),
+		EventRecorder:         eventRecorder,
+		AdditionalAnnotations: additionalAnnotations,
+		Owner:                 owner,
+		UseSecretUpdateOnly:   true,
+	}
+
+	// Ensure we don't leak goroutines
+	initialNumGoroutine := runtime.NumGoroutine()
+
+	c := NewCertRotationControllerMultipleTargets("operator", signerSecret, caBundleConfigMap, []RotatedSelfSignedCertKeySecret{targetFirstSecret, targetSecondSecret}, eventRecorder, &fakeStatusReporter{})
+
+	informerFactory.Start(controllerCtx.Done())
+	hasSecretSynced := cache.WaitForCacheSync(controllerCtx.Done(), informerFactory.Core().V1().Secrets().Informer().HasSynced)
+	if hasSecretSynced != true {
+		t.Errorf("caches for secrets didn't sync")
+	}
+	hasConfigMapsSynced := cache.WaitForCacheSync(controllerCtx.Done(), informerFactory.Core().V1().ConfigMaps().Informer().HasSynced)
+	if hasConfigMapsSynced != true {
+		t.Errorf("caches for configmap didn't sync")
+	}
+
+	time.AfterFunc(1*time.Second, func() {
+		cancel()
+	})
+	firstTargetCertCreator.triggerSync()
+	secondTargetCertCreator.triggerSync()
+	c.Run(controllerCtx, 1)
+
+	// Ensure both target certs have been called exactly three times
+	// initial sync and two hook calls for target certs
+	// TODO[vrutkovs]: informers make unpredictable number of calls
```

Comment:
Not sure how to tackle that - or how to make sure two hook syncs were included in regular informer sync. `informerFactory` and `NewCertRotationControllerMultipleTargets` promise to sync every minute, but it happens much more often